plugins/catalog-backend/src/service/NextRouter.ts

```diff
@@ -70,7 +71,13 @@ export async function createNextRouter(
   if (refreshService) {
     router.post('/refresh', async (req, res) => {
       const refreshOptions: RefreshOptions = req.body;
-      await refreshService.refresh(refreshOptions);
+      const authToken = getAuthToken(req);
+      if (!authToken) {
```

Comment:
I feel this should be up to the `AuthorizedRefreshService` instead, throwing an error instead. Especially since the token is optional in the interface

Reply:
If we take this out and leave the rest as-is, we can actually delegate the handling of the undefined token case to the permission client (and in turn, the permission backend), which I think is sensible.